src/neo/SmartContract/Native/Tokens/NeoToken.Economic.cs

```diff
@@ -0,0 +1,165 @@
+        {
+            var endRewardItem = snapshot.Storages.TryGet(CreateStorageKey(Prefix_HolderRewardPerBlock).Add(uint.MaxValue - end - 1));
+            var startRewardItem = snapshot.Storages.TryGet(CreateStorageKey(Prefix_HolderRewardPerBlock).Add(uint.MaxValue - start - 1));
+            BigInteger startReward = startRewardItem is null ? 0 : new BigInteger(startRewardItem.Value);
+            return value * (new BigInteger(endRewardItem.Value) - startReward) / TotalAmount;
+        }
+
+        [ContractMethod(0_03000000, CallFlags.AllowStates)]
+        public BigInteger UnclaimedGas(StoreView snapshot, UInt160 account, uint end)
+        {
+            StorageItem storage = snapshot.Storages.TryGet(CreateStorageKey(Prefix_Account).Add(account));
+            if (storage is null) return BigInteger.Zero;
+            NeoAccountState state = storage.GetInteroperable<NeoAccountState>();
+            return CalculateBonus(snapshot, state.VoteTo, state.Balance, state.BalanceHeight, end);
+        }
+
+        private void DistributeGasForCommittee(ApplicationEngine engine)
+        {
+            var gasPerBlock = GetGasPerBlock(engine.Snapshot);
+            (ECPoint, BigInteger)[] committeeVotes = GetCommitteeVotes(engine.Snapshot);
+            int validatorNumber = GetValidators(engine.Snapshot).Length;
+            RewardRatio rewardRatio = GetRewardRatio(engine.Snapshot);
+            BigInteger holderRewardPerBlock = gasPerBlock * rewardRatio.NeoHolder / 100; // The final calculation should be divided by the total number of NEO
+            BigInteger committeeRewardPerBlock = gasPerBlock * rewardRatio.Committee / 100 / committeeVotes.Length;
+            BigInteger voterRewardPerBlock = gasPerBlock * rewardRatio.Voter / 100 / (committeeVotes.Length + validatorNumber);
+
+            // Keep track of incremental gains of neo holders
+
+            var index = engine.Snapshot.PersistingBlock.Index;
+            var holderRewards = holderRewardPerBlock;
+            var holderRewardKey = CreateStorageKey(Prefix_HolderRewardPerBlock).Add(uint.MaxValue - index - 1);
+            var holderBorderKey = CreateStorageKey(Prefix_HolderRewardPerBlock).Add(uint.MaxValue);
+            var enumerator = engine.Snapshot.Storages.FindRange(holderRewardKey, holderBorderKey).GetEnumerator();
+            if (enumerator.MoveNext())
+                holderRewards += new BigInteger(enumerator.Current.Value.Value);
+            engine.Snapshot.Storages.Add(holderRewardKey, new StorageItem() { Value = holderRewards.ToByteArray() });
+
+            for (var i = 0; i < committeeVotes.Length; i++)
+            {
+                // Keep track of incremental gains for each committee's voters
+
+                UInt160 committeeAddr = Contract.CreateSignatureContract(committeeVotes[i].Item1).ScriptHash;
+                if (committeeVotes[i].Item2 > 0)
```

Comment:
`StandbyCommittee` will initially have no votes. I'm not sure if it is ok. _(After confirmed, I will fix UT)_